graylog2-server/src/main/java/org/graylog2/security/ldap/LdapSettingsImpl.java

```diff
@@ -238,4 +248,54 @@ public void setTrustAllCertificates(boolean trustAllCertificates) {
         fields.put(TRUST_ALL_CERTS, trustAllCertificates);
     }
 
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    @Override
+    public Map<String, String> getGroupMapping() {
+        final Map<String, String> groupMapping = (Map<String, String>) fields.get(GROUP_MAPPING);
+
+        if (groupMapping == null || groupMapping.isEmpty()) {
+            return Maps.newHashMap();
+        }
+        else {
+            // we store role ids, but the outside world uses role names to identify them
+            try {
+                final Map<String, Role> idToRole = roleService.loadAllIdMap();
+                return Maps.newHashMap(Maps.transformValues(groupMapping, new Role.RoleIdToNameFunction(idToRole)));
+            } catch (NotFoundException e) {
+                LOG.error("Unable to load role mapping");
+                return Maps.newHashMap();
```

Comment:
Shouldn't this actually throw an exception or at least somehow indicate that loading the group mapping failed? Otherwise users might think that the call succeeded if they don't check the error log.